js/src/redux/providers/registry/middleware.js

```diff
@@ -14,13 +14,40 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
+import { debounce } from 'lodash';
 import Contracts from '~/contracts';
 import subscribeToEvents from '~/util/subscribe-to-events';
 
 import registryABI from '~/contracts/abi/registry.json';
 
 import { setReverse, startCachingReverses } from './actions';
 
+const read = (chain) => {
+  const reverses = window.localStorage.getItem(`${chain}-registry-reverses`);
+  const lastBlock = window.localStorage.getItem(`${chain}-registry-reverses-last-block`);
+  if (!reverses || !lastBlock) {
+    return null;
+  }
+
+  try {
+    return {
+      reverses: JSON.parse(reverses),
+      lastBlock: JSON.parse(lastBlock)
+    };
+  } catch (_) {
+    return null;
+  }
+};
+
+const write = debounce((getChain, getReverses, getLastBlock) => {
+  const chain = getChain();
+  const reverses = getReverses();
+  const lastBlock = getLastBlock();
+
+  window.localStorage.setItem(`${chain}-registry-reverses`, JSON.stringify(reverses));
```

Comment:
Additional bonus for using `store` (as commented above) - no need to manually strigify